From 1a529fcfd74aad7e1182158703e64b48c74bc602 Mon Sep 17 00:00:00 2001
From: gaopan <b54642@freescale.com>
Date: Tue, 12 May 2015 18:25:29 +0800
Subject: [PATCH 0025/5242] MLK-10893: i2c: imx: add irqf_no_suspend

commit  2e003654c7deeb562e7dd913fc56303555d142c5 from
https://source.codeaurora.org/external/imx/linux-imx.git

The i2c irq is masked when pcie starts a i2c transfer process
during noirq suspend stage. As a result, i2c transfer fails.
To solve the problem, IRQF_NO_SUSPEND is added to i2c bus.

Signed-off-by: Gao Pan <b54642@freescale.com>
Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/i2c/busses/i2c-imx.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-imx.c b/drivers/i2c/busses/i2c-imx.c
index ad6adef..1d8c5f1 100644
--- a/drivers/i2c/busses/i2c-imx.c
+++ b/drivers/i2c/busses/i2c-imx.c
@@ -1102,7 +1102,8 @@ static int i2c_imx_probe(struct platform_device *pdev)
 	}
 
 	/* Request IRQ */
-	ret = devm_request_irq(&pdev->dev, irq, i2c_imx_isr, IRQF_SHARED,
+	ret = devm_request_irq(&pdev->dev, irq, i2c_imx_isr,
+				IRQF_SHARED | IRQF_NO_SUSPEND,
 				pdev->name, i2c_imx);
 	if (ret) {
 		dev_err(&pdev->dev, "can't claim irq %d\n", irq);
-- 
1.7.9.5

